hoptimator-util/src/main/java/com/linkedin/hoptimator/util/planner/PipelineRel.java

```diff
@@ -78,7 +90,26 @@ public void setSink(String database, List<String> path, RelDataType rowType, Map
       this.sinkDatabase = database;
       this.sinkPath = path;
       this.sinkRowType = rowType;
-      this.sinkOptions = options;
+      this.sinkOptions = addKeysAsOption(options, rowType);
+    }
+
+    private Map<String, String> addKeysAsOption(Map<String, String> options, RelDataType rowType) {
+      Map<String, String> newOptions = new HashMap<>(options);
```

Comment:
Probably should make these LinkedHashMaps to preserve order. The integration tests expect deterministic output.